### DISCUSSION TITLE
HLS Stream

### DISCUSSION THREAD

**Discussion by ariadnalls** [HaishinKit/HaishinKit.swift#1261] [Q&A]
Hi I've been using the library for streaming RTMP and it's working great so far.

I'm trying to stream using HLS to Youtube but they only provide an url and a key.
How should I create the http service then?

Thank you

**Comment by shogo4405** [comment]
I haven't designed it for purposes like forwarding to other servers like YouTube, so unfortunately, that isn't possible.

**Reply by ariadnalls** [reply]
Any plans on adding it? It would be really useful

https://support.google.com/youtube/answer/10349430?hl=en&visit_id=638271653467121903-3580700581&p=hls_stream&rd=1

<img width="447" alt="image" src="https://github.com/shogo4405/HaishinKit.swift/assets/6066266/43b21d59-4694-42b7-81ad-bcff0924d5cb">

Yo can do so by Select stream key > Create new stream key > Streaming protocol > HLS (Advanced)


Thank you!

**Reply by shogo4405** [reply]
I would like to know for reference, but it seems that there isn't any particular advantage of using HLS over RTMP for forwarding. Why is HLS delivery necessary in this case?

**Reply by amudhansekar** [reply]
Hi, I am also making a program for school and wanted to use your swift kit to feed into the youtube rtmp or hls server. The rtmp stream  give the response data"streamName": "XXXX-XXXX-XXXX-XXXX-XXXX-XXXX", "ingestionAddress": "rtmp://a.rtmp.youtube.com/live2", and the hls response data is "streamName": "XXXX-XXXX-XXXX-XXXX-XXXX-XXXX",
      "ingestionAddress": "https://a.upload.youtube.com/http_upload_hls?cid=XXXX-XXXX-XXXX-XXXX-XXXX-XXXX&copy=0&file=", Is it possible I can just forward the playlist.m3u8 into the ingestion Address? I tried variations of the URL but I have hard time of knowing the exact URL that is being made. I don't have a server on my own, but it seems like I don't need it because your code encodes the camera data. Please let me know your thoughts and thanks again for making this your saving me in my CS class :)

**Reply by shogo4405** [reply]
@amudhansekar I state that there is no functionality to ingest to YouTube using HLS. Since it's not possible, please give up on it.

**Reply by amudhansekar** [reply]
Is there functionality to ingest to youtube using rtmp? Is it possible to ingest without making my own rtmp server?
